### ISSUE TITLE
Request for code samples to be licensed under MIT license

### ISSUE THREAD

**Issue by mmucklo** [symfony/symfony-docs#12124]
This is a request for the code samples included in the documentation to be licensed under the same terms as the symfony source code.

1) Confusion

It's confusing to have two licenses - one for the source code you're allowed to use, and one for the sample code you're supposed to copy.

2) CC-By-SA 3.0 not recommended for source code.

[Can I apply a Creative Commons license to software?](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software)

Please at least use an OSI license.

3) Attribution required

The present code-sample license requests that attribution is required.

That would mean if I build a symfony-website, I would need to have some kind of footer giving attribution back to symfony.com on every page. This is much more restrictive than the symfony source code, and could be considered quite tacky to those having to do this for symfony freelance work.

===
In contrast maybe you could sue everyone who's now using symfony sample code without attribution for license violation, if that's your intent (or are secretly reserving the right to do so as a nuclear option), however it would be nice to know that up front.

**Comment by javiereguiluz** [comment]
First, to state things perfectly clear: nobody is going to sue anybody; there's no "nuclear option"; you don't have to give any attribution to Symfony in any way in any app created with Symfony.

-----

We consider that the code included in the Symfony Docs are "code examples" and part of the docs, not "source code" part of a software application. We apply a single license to anything related to docs: words, images and code examples.

And if someone still wonders *"yes, but what if?*:

* Code examples are incomplete snippets. You can't build a real app copy+pasting these snippets.
* Even if you copy some lines or some code snippets ... you'll change them. For example, to rename the variables or complete the code.
* In the imaginary case that you build a real (but private/commercial) app only copy+pasting these snippets ... we can't know in any way that your app is built with Symfony because we don't have access to its source code.

In summary: any Symfony developer is perfectly safe and it's OK to copy+paste code examples and adapt them in your app.

I hope things are more clear now. Thanks!
